src/server/game/Spells/SpellEffects.cpp

```diff
@@ -273,6 +276,15 @@ void Spell::EffectResurrectNew(SpellEffIndex effIndex)
     SendResurrectRequest(target);
 }
 
+void DelayMe(float seconds)
+{
+    std::clock_t startClock = std::clock();
+    float secondsAhead = seconds * CLOCKS_PER_SEC;
+    // do nothing until the elapsed time has passed.
+    while (std::clock() < startClock + secondsAhead);
+    return;
+}
```

Comment:
```suggestion
void DelayMe(Milliseconds delayMs)
{
    SystemTimePoint delay = std::chrono::system_clock::now() + delayMs;

    // do nothing until the elapsed time has passed.
    while (std::chrono::system_clock::now() < delay);

    return;
}
```

Reply:
Nice catch, for a second I totally forgot about chrono